### PR TITLE
Change: Widget layout tweaks

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Some windows have layouts with widgets that overlap panels, which can look a little odd and makes applying uniform padding
problematic.

## Description

This PR features minor tweaks to widget layouts and a couple of other extras:

1. Use of zero-sized pane in the world gen UI means rescaling this window can have the wrong result, instead an spacer widget is used.
2. File load/save windows have panels that overlap other panels. This predates the nested widget system.
3. Simplified widget tree for story, settings, industry-cargo and graph legend windows.
4. AI Competitor limit button size set to same size as [<][>] "Arrow Buttons" as used elsewhere.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by PeterN** [comment]
Old file window
image

New file window 
image

File: src/genworld_gui.cpp [hunk 1]
```diff
@@ -499,10 +504,10 @@ struct GenerateLandscapeWindow : public Window {
...
 		int climate_plane = 0;
 		switch (_settings_newgame.game_creation.landscape) {
-			case LT_TEMPERATE: climate_plane = SZSP_VERTICAL; break;
-			case LT_ARCTIC:    climate_plane = 0;             break;
-			case LT_TROPIC:    climate_plane = 1;             break;
-			case LT_TOYLAND:   climate_plane = SZSP_VERTICAL; break;
+			case LT_TEMPERATE: climate_plane = 2; break;
+			case LT_ARCTIC:    climate_plane = 0; break;
+			case LT_TROPIC:    climate_plane = 1; break;
+			case LT_TOYLAND:   climate_plane = 2; break;
```
Comment (LordAro) [inline src/genworld_gui.cpp:510]: new magic numbers?

Comment (PeterN) [reply src/genworld_gui.cpp:510]: It's the just the number of the plane, which the index of the child widgets of the stacked plane. We don't have defines for these anywhere else either, the only case is when we use the `SZSP_` constants which are magic numbers.

File: src/story_gui.cpp [hunk 2]
```diff
@@ -970,19 +970,15 @@ static const NWidgetPart _nested_story_book_widgets[] = {
...
-			NWidget(NWID_VSCROLLBAR, COLOUR_BROWN, WID_SB_SCROLLBAR),
-			NWidget(WWT_RESIZEBOX, COLOUR_BROWN),
-		EndContainer(),
+		NWidget(WWT_PANEL, COLOUR_BROWN, WID_SB_PAGE_PANEL), SetResize(1, 1), SetScrollbar(WID_SB_SCROLLBAR), EndContainer(),
+		NWidget(NWID_VSCROLLBAR, COLOUR_BROWN, WID_SB_SCROLLBAR),
+	EndContainer(),
+	NWidget(NWID_HORIZONTAL),
+		NWidget(WWT_TEXTBTN, COLOUR_BROWN, WID_SB_PREV_PAGE), SetMinimalSize(100, 0), SetFill(0, 0), SetDataTip(STR_STORY_BOOK_PREV_PAGE, STR_STORY_BOOK_PREV_PAGE_TOOLTIP),
+		NWidget(NWID_BUTTON_DROPDOWN, COLOUR_BROWN, WID_SB_SEL_PAGE), SetMinimalSize(93, 12), SetFill(1, 0),
+												SetDataTip(STR_BLACK_RAW_STRING, STR_STORY_BOOK_SEL_PAGE_TOOLTIP), SetResize(1, 0),
```
Comment (LordAro) [inline src/story_gui.cpp:979]: Unusual amount of indentation for this line... (not that it was added by you.. (in this PR, anyway))

**Review by LordAro — APPROVED** [review]

**Pull request merged**